compiler/semantic/op.go

```diff
@@ -83,6 +83,21 @@ func (a *analyzer) semTrunk(trunk ast.Trunk, out dag.Seq) (dag.Seq, error) {
 
 //XXX make sure you can't read files from a lake instance
 
+func (a *analyzer) semOpSource(source ast.Source, out dag.Seq) (dag.Seq, error) {
+	sources, err := a.semSource(source)
+	if err != nil {
+		return nil, err
+	}
+	if len(sources) == 1 {
+		return append(out, append(dag.Seq{sources[0]}, out...)...), nil
+	}
+	paths := make([]dag.Seq, len(sources))
+	for k, s := range sources {
+		paths[k] = dag.Seq{s}
```

Comment:
Minor nit: I think loops like this are a little clearer with `var` and `append`.
```suggestion
	var paths []dag.Seq
	for _, s := range sources {
		paths = append(paths, dag.Seq{s})
```